plugins/dell-kestrel/fu-dell-kestrel-ec.c

```diff
@@ -440,143 +443,141 @@ fu_dell_kestrel_ec_own_dock(FuDevice *device, gboolean lock, GError **error)
 }
 
 gboolean
-fu_dell_kestrel_ec_run_passive_update(FuDevice *device, GError **error)
+fu_dell_kestrel_ec_run_passive_update(FuDellKestrelEc *self, GError **error)
 {
 	g_autoptr(GByteArray) req = g_byte_array_new();
-	g_return_val_if_fail(device != NULL, FALSE);
 
 	/* ec included in cmd, set bit2 in data for tbt */
 	fu_byte_array_append_uint8(req, FU_DELL_KESTREL_EC_HID_CMD_SET_PASSIVE);
 	fu_byte_array_append_uint8(req, 1); // length of data
 	fu_byte_array_append_uint8(req, 0x02);
 
 	g_debug("registered passive update (uod) flow");
-	return fu_dell_kestrel_ec_write(device, req, error);
+	return fu_dell_kestrel_ec_write(self, req, error);
 }
 
 static gboolean
-fu_dell_kestrel_ec_set_dock_sku(FuDevice *device, GError **error)
+fu_dell_kestrel_ec_set_dock_sku(FuDellKestrelEc *self, GError **error)
 {
-	FuDellKestrelEc *self = FU_DELL_KESTREL_EC(device);
+	if (self->base_type == FU_DELL_DOCK_BASE_TYPE_KESTREL) {
+		g_autoptr(FuStructDellKestrelDockInfoEcQueryEntry) dev_entry = NULL;
 
-	switch (self->base_type) {
-	case FU_DELL_DOCK_BASE_TYPE_KESTREL:
 		/* TBT type yet available, do workaround */
-		if (fu_dell_kestrel_ec_dev_entry(device,
-						 FU_DELL_KESTREL_EC_DEV_TYPE_TBT,
-						 FU_DELL_KESTREL_EC_DEV_SUBTYPE_BR,
-						 0) != NULL) {
+		dev_entry = fu_dell_kestrel_ec_dev_entry(self,
+							 FU_DELL_KESTREL_EC_DEV_TYPE_TBT,
+							 FU_DELL_KESTREL_EC_DEV_SUBTYPE_BR,
+							 0);
+		if (dev_entry != NULL) {
 			self->base_sku = FU_DELL_KESTREL_DOCK_SKU_T5;
 			return TRUE;
 		}
-		if (fu_dell_kestrel_ec_dev_entry(device,
-						 FU_DELL_KESTREL_EC_DEV_TYPE_TBT,
-						 FU_DELL_KESTREL_EC_DEV_SUBTYPE_GR,
-						 0) != NULL) {
+		dev_entry = fu_dell_kestrel_ec_dev_entry(self,
+							 FU_DELL_KESTREL_EC_DEV_TYPE_TBT,
+							 FU_DELL_KESTREL_EC_DEV_SUBTYPE_GR,
+							 0);
+		if (dev_entry != NULL) {
 			self->base_sku = FU_DELL_KESTREL_DOCK_SKU_T4;
 			return TRUE;
 		}
 		self->base_sku = FU_DELL_KESTREL_DOCK_SKU_DPALT;
 		return TRUE;
-	default:
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_NOT_FOUND,
-			    "unsupported dock type: %x",
-			    self->base_type);
-		return FALSE;
 	}
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_NOT_FOUND,
+		    "unsupported dock type: %x",
+		    self->base_type);
+	return FALSE;
 }
 
 guint32
 fu_dell_kestrel_ec_get_pd_version(FuDevice *device,
```

Comment:
do we want `FuDellKestrelEc` in the parameter field?

Reply:
Yes, please. Please fix up any more you see -- thanks!